exercises/raindrops/src/example.c

```diff
@@ -15,7 +15,7 @@ char *convert(char result[], int drops)
       strcat(result, "Plong");
    }
    if (strlen(result) == 0) {
-      char drops_string[12] = "";
+      char drops_string[12] = "\0";
```

Comment:
Isn't this equivalent? What was the warning that this fixes?

Reply:
The error given is for ln #19 (`strcat(result, drops_string)`). It reads `(error) Buffer is accessed out of bounds`.

Of course the buffer is _not_ accessed out of bounds by the example code. This then means that either the standard library functions used with the buffer directly reference some out of bounds index or this is a false positive. I find the former hard to credit. While searching online on this I came across [this sourcforge bug report](https://sourceforge.net/p/cppcheck/discussion/general/thread/2c33dfc5/) that mentions a similar false positive in the initialization. The fix for that made it into CppCheck 1.81, but the version in the Ubuntu PPA is 1.72. Getting a later version requires building from source so I've not been able to see if a later version fixes this due to time constraints (I have actually had a stab at this but it turned into dependency-fest).

While `= ""` should work, `= "\0";`, `= { '\0' }` or `= { 0 };` are all equally as valid. I'm not sure on the C spec for initialization of empty strings but I do know that for initializer *lists* it mentions something like 

> An initializer-list must have at least one initializer

and also 

> If there are fewer initializers in a brace-enclosed list than there are elements or members of an aggregate, or fewer characters in a string literal used to initialize an array of known size than there are elements in the array, the remainder of the aggregate shall be initialized implicitly the same as objects that have static storage duration

`"\0"` seems to fit the bill with the least amount of change.

Reply:
Ah, sounds good. Thanks for the in-depth explanation!